src/components/SelectedSurveyPage.jsx

```diff
@@ -26,6 +26,19 @@ export default class SelectedSurveyPage extends PureComponent {
     fetchSurvey(id);
   }
 
+  // TODO(ayushbhagat): Check for cycles.
```

Comment:
Yes, the backend should do all validations of the survey and respond with an error if it's invalid in any way.